File: 프로그래머스/0/120812. 최빈값 구하기/최빈값 구하기.js
function solution(array) {
    const map = new Map();
    
    for(const num of array) {
        map.set(num, (map.get(num) || 0) + 1);
    }
    
    const array2 = [...map].sort((a, b) => b[1] - a[1]);
    return array2.length === 1 || array2[0][1] > array2[1][1] ? array2[0][0] : -1;
}

File: 프로그래머스/0/120812. 최빈값 구하기/최빈값 구하기.java
import java.util.*;

class Solution {
    public int solution(int[] array) {
        Map<Integer, Integer> countMap = new HashMap<>();
        
        for (int num : array) {
            countMap.put(num, countMap.getOrDefault(num, 0) + 1);
        }
        
        int mode = -1;
        int maxCount = 0;
        boolean isDuplicated = false;
        
        for(Map.Entry<Integer, Integer> entry : countMap.entrySet()) {
            int value = entry.getKey();
            int count = entry.getValue();
            
            if(count > maxCount) {
                maxCount = count;
                mode = value;
                isDuplicated = false;
            } else if(count == maxCount) {
                isDuplicated = true;
            }
        }
        
        return isDuplicated ? -1 : mode;
    }
}